### ISSUE TITLE
Query parameters stopped working in the WITH FILL modifier

### ISSUE THREAD

**Issue by alexey-milovidov** [ClickHouse/ClickHouse#62113]
```
$ clickhouse-local --param_rounding 1 --query "SELECT {rounding:UInt32}"
1
$ clickhouse-local --param_rounding 1 --query "SELECT 1 AS x ORDER BY x WITH FILL STEP {rounding:UInt32}"
Code: 456. DB::Exception: Query parameter `rounding` was not set. (UNKNOWN_QUERY_PARAMETER)

$ programs/server/clickhouse-24.1 local --param_rounding 1 --query "SELECT 1 AS x ORDER BY x WITH FILL STEP {rounding:UInt32}"
1
```

**Comment by biggerfisch** [comment]
I can reproduce this with the docker versions too; makes `24.3` unusable for us, specifically `24.3.1.2672`

**Comment by novikd** [comment]
@biggerfisch There is a bug in query parameters implementation and I'm currently working on a fix: #62186. Until it's fixed I offer you to disable `allow_experimental_analyzer` as a workaround.
